src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -64,31 +64,39 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     throw CreateSslException(SR.net_allocate_ssl_context_failed);
                 }
 
-                // TLS 1.3 uses different ciphersuite restrictions than previous versions.
-                // It has no equivalent to a NoEncryption option.
-                if (policy == EncryptionPolicy.NoEncryption)
+                if (!Interop.Ssl.Tls13Supported)
+                {
+                    if (protocols != SslProtocols.None &&
+                        CipherSuitesPolicy.WantsTls13(protocols))
+                    {
+                        protocols = protocols & (~SslProtocols.Tls13);
+                    }
+                }
+                else if (CipherSuitesPolicy.WantsTls13(protocols) &&
+                    CipherSuitesPolicy.ShouldOptOutOfTls13(sslAuthenticationOptions.CipherSuitesPolicy, policy))
                 {
                     if (protocols == SslProtocols.None)
                     {
+                        // we are using default settings but cipher suites policy says that TLS 1.3
+                        // is not compatible with our settings (i.e. we requested no encryption or disabled
+                        // all TLS 1.3 cipher suites)
                         protocols = SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12;
                     }
                     else
                     {
-                        protocols &= ~SslProtocols.Tls13;
-
-                        if (protocols == SslProtocols.None)
-                        {
-                            throw new SslException(
-                                SR.Format(SR.net_ssl_encryptionpolicy_notsupported, policy));
-                        }
+                        // user explicitly asks for TLS 1.3 but their policy is not compatible with TLS 1.3
+                        throw new SslException(
+                            SR.Format(SR.net_ssl_encryptionpolicy_notsupported, policy));
```

Comment:
@bartonjs actually this happens when user wants TLS1.3 (protocols include it) but user specified ciphersuites in such a way that TLS1.3 cannot happen (none TLS 1.3 ciphersuites included in a policy) - so user contradicted themselves

Reply:
>  so user contradicted themselves

Yeah, I know.  The cases where this would come up seem vanishingly small, but the easy one is trying to build something like SslLabs, where you connect, see what ciphersuite got negotiated, remove it from the allow list, and try again.  When the last TLS 1.3 ciphersuite gets removed the next call throws.

The throw just means that we're pushing the knowledge of TLS 1.3 vs prior ciphersuites onto the caller.  Which, given we are encouraging people to just use system defaults (after configuring their system, if they don't like the out-of-the-box answer), seems okay for now.

Reply:
I got mixed feelings for that - if user doesn't want to know about TLS 1.3 they should use None for protocols in which case we won't throw.

Perhaps let's leave as is and perhaps change it when someone complains this is not intuitive?

I don't feel strongly either way - allowing this would simplify a code a bit but feels to me that asking for TLS 1.3 and not passing any ciphers which could allow us to do so is a user bug - for scenarios like SslLabs they will have to understand the cipher suites either way...